pkg/capacityscheduling/capacity_scheduling.go

```diff
@@ -347,6 +407,87 @@ func (c *CapacityScheduling) preempt(ctx context.Context, state *framework.Cycle
+// considered for preemption.
+// We look at the node that is nominated for this pod and as long as there are
+// terminating pods on the node, we don't consider this for preempting more pods.
+func (c *CapacityScheduling) PodEligibleToPreemptOthers(pod *v1.Pod, nodeInfos framework.NodeInfoLister, nominatedNodeStatus *framework.Status, state *framework.CycleState) bool {
+	if pod.Spec.PreemptionPolicy != nil && *pod.Spec.PreemptionPolicy == v1.PreemptNever {
+		klog.V(5).Infof("Pod %v/%v is not eligible for preemption because it has a preemptionPolicy of %v", pod.Namespace, pod.Name, v1.PreemptNever)
+		return false
+	}
+
+	preFilterState, err := getPreFilterState(state)
+	if err != nil {
+		klog.Errorf("error reading %q from cycleState: %v", preFilterStateKey, err)
+		return false
+	}
+
+	nomNodeName := pod.Status.NominatedNodeName
+	if len(nomNodeName) > 0 {
+		// If the pod's nominated node is considered as UnschedulableAndUnresolvable by the filters,
+		// then the pod should be considered for preempting again.
+		if nominatedNodeStatus.Code() == framework.UnschedulableAndUnresolvable {
+			return true
+		}
+
+		elasticQuotaSnapshotState, err := getElasticQuotaSnapshotState(state)
+		if err != nil {
+			klog.Errorf("error reading %q from cycleState: %v", ElasticQuotaSnapshotKey, err)
+			return true
+		}
+		preemptorElasticQuotaInfo, preemptorWithElasticQuota := elasticQuotaSnapshotState.elasticQuotaInfos[pod.Namespace]
+
+		var moreThanMinWithPreemptor bool
+		// Check if there is elastic quota in the preemptor's namespace.
+		if preemptorWithElasticQuota {
+			moreThanMinWithPreemptor = preemptorElasticQuotaInfo.overUsed(preFilterState.nominatedPodsReqWithPodReq, preemptorElasticQuotaInfo.Min)
+		}
+		nodeInfo, _ := nodeInfos.Get(nomNodeName)
+		if nodeInfo == nil {
+			return true
+		}
+
+		podPriority := corev1helpers.PodPriority(pod)
+		if preemptorWithElasticQuota {
+			for _, p := range nodeInfo.Pods {
+				if p.Pod.DeletionTimestamp != nil {
+					pWithElasticQuotaInfo, pWithElasticQuota := elasticQuotaSnapshotState.elasticQuotaInfos[p.Pod.Namespace]
+					if !pWithElasticQuota {
+						continue
+					}
+					if p.Pod.Namespace == pod.Namespace && corev1helpers.PodPriority(p.Pod) < podPriority {
+						// There is a terminating pod on the nominated node.
+						// If the terminating pod is in the same namespace with preemptor
+						// and it is less important than preemptor,
+						// return false to avoid preempting more pods.
+						return false
+					} else if p.Pod.Namespace != pod.Namespace && !moreThanMinWithPreemptor && pWithElasticQuotaInfo.usedOverMin() {
+						// There is a terminating pod on the nominated node.
+						// The terminating pod isn't in the same namespace with preemptor.
+						// If moreThanMinWithPreemptor is false, it indicates that preemptor can preempt the pods belong to the namespace whose quota is over used than min.
+						return false
```

Comment:
Why return false?

Reply:
I add more comments in the code.
1. The terminating pod isn't in the same namespace with preemptor.
2. If moreThanMinWithPreemptor is false, it indicates that preemptor can preempt the pods belong to the namespace whose quota is over used than min.
3. And if the used of terminating pod's quota is over min, so the space released by terminating pod on the nominated node can be used by the preemptor.
4. return false to avoid preempting more pods.

Reply:
thanks for the explanation.

```
// If moreThanMinWithPreemptor is false, it indicates that preemptor can preempt the pods belong to the namespace whose quota is over used than min.
```

Can we reword to:

```
// If moreThanMinWithPreemptor is false, it indicates that preemptor can preempt the pods in other EQs whose used is over min.
```